client/src/hooks/useHash.ts

```diff
@@ -0,0 +1,32 @@
+import { useMemo, useState, useEffect } from "react";
+
+import { useParams } from "next/navigation";
+
+export const useHash = () => {
+  const getCurrentHash = useMemo(
+    () => () => (typeof window !== "undefined" ? window.location.hash : ""),
```

Comment:
Given the hook will run in client side, I think the window check is not necessary.